src/memory/mod.rs

```diff
@@ -65,11 +65,13 @@
 //! responsible for reconstructing the `UnifiedBuffer` using `from_raw_parts()` and dropping it to
 //! ensure that the memory allocation is safely cleaned up.
 
+mod array;
 mod device;
 mod locked;
 mod malloc;
 mod unified;
 
+pub use self::array::*;
```

Comment:
I think the array types could go in their own module, rather than adding still more stuff to memory.